src/index/IndexMetaData.h

```diff
@@ -163,6 +169,13 @@ using IndexMetaDataMmapView = IndexMetaData<
 // constants for Magic Numbers to separate different types of MetaData;
 const size_t MAGIC_NUMBER_MMAP_META_DATA = static_cast<size_t>(-1);
 const size_t MAGIC_NUMBER_SPARSE_META_DATA = static_cast<size_t>(-2);
-const size_t MAGIC_NUMBER_MMAP_META_DATA_BLOCK_LIST = static_cast<size_t>(-3);
```

Comment:
where is this gone? The version is still called `V_BLOCK_LIST_AND_STATISTICS`

Reply:
Yes, I would suggest a magic number that stays fixed from now on and a version Tag which can be incremented. The magic number must still discriminate whether we have a version tag at all, and thus we have 4 Magic Numbers for sparse/dense and with/without version.

The version tag describes the features of this version (thus "block list" and "statistics").

In one of the next PRs I will clean the serialization/deserialization of the IndexMetaData a bit more, e.g. only have one createFromByteBuffer that handles sparse and dense meta data using constexpr if etc.

Reply:
I thought wherever we added a magic number we also added a version field? If not maybe we should just drop support for those files where we didn't do that?

Reply:
Okay, this is a design decision:
1. Dropping the support makes the code easier
2. Dropping the support breaks ALL Indices created so far
3. We could also archive the old versions of createFromByteBuf somewhere for the MetaDataConverter. We would then have some code duplications, but somewhere in a frozen state far away from the actual production code.

Just tell me what you like best.

Reply:
I mean didn't we add a version wherever we added a magic number? So that we only got two formats
* No magic number and no version = legacy format
* Magic number and version = can increment version and keep magic number

Reply:
Unfortunately we forgot to add a version with the IndexMetaData.
But I think we could implement your suggestion and we would break only the Indices built in the last days, using the "MMapBasedVector"-PR. I think this only affects your freebase builds which were converted from old builds and could be converted again. Then we would only have ONE magic number for sparse/dense each and the rest is handled by a version tag which I consider to be cleaner. What do you think of this?

Reply:
I converted most of our QLever indices already and @hannahbast  built a new version of Wikipedia + Freebase Easy but that's ok. I'd rather we break some intermediate indices now than be left with a worse design later.

Reply:
Ok, I think I finally got you now.( I first implemented a version that can read ANYTHING). What I can do with this design is Inform the User, that his index can not be fixed (should only affect you and Prof.Bast)